sdk/src/main/java/velocitas/sdk/grpc/GrpcCall.kt

```diff
@@ -0,0 +1,76 @@
+package velocitas.sdk.grpc
+
+import io.grpc.Channel
+import io.grpc.Grpc
+import io.grpc.Status
+import io.grpc.stub.StreamObserver
+import org.eclipse.kuksa.proto.v1.BrokerGrpc
+import org.eclipse.kuksa.proto.v1.BrokerOuterClass.SubscribeReply
+import org.eclipse.kuksa.proto.v1.BrokerOuterClass.SubscribeRequest
+import velocitas.sdk.Logger
+
+abstract class GrpcCall {
+    // grpc::ClientContext context
+    var isComplete = false
+}
+
+/**
+ * A GRPC call where a request is followed up by a single reply.
+ *
+ * @param TRequestType The data type of the request.
+ * @param TReplyType   The data type of the reply.
+ */
+class GrpcSingleResponseCall<TRequestType, TReplyType> : GrpcCall() {
+    var request: TRequestType? = null
+    var reply: TReplyType? = null
+}
+
+class GrpcStreamingResponseCall<TRequestType, TReplyType> : GrpcCall() {
+
+    var request: TRequestType? = null
+        private set
+
+    private val readReactor: ReadReactor<TRequestType, TReplyType> = ReadReactor(this)
```

Comment:
Fixed. A bigger refactoring was done here - I think it should be fine, since it mostly contains internal / grpc specific stuff. 
I don't exactly know why we need the GrpcCall, because basically the only thing is does, is tracking the active requests / responses and the complete state for which is tracked inside the GrpcClient. But let's keep it for now (maybe something to be discussed inside the Architecture round, what's the purpose of this thingy)

It's still pretty rough around the edges, but all in all I'm a satisfied how the GrpcCall / AsyncBrokerGrpcFacade now handles the logic.